util/headerreader/header_reader.go

```diff
@@ -88,13 +91,22 @@ var TestConfig = Config{
 }
 
 func New(client arbutil.L1Interface, config ConfigFetcher) *HeaderReader {
+	isParentChainArbitrum := false
+	arbSys, err := precompilesgen.NewArbSys(types.ArbSysAddress, client)
+	if err == nil {
+		_, err := arbSys.ArbBlockNumber(&bind.CallOpts{Pending: true})
+		if err == nil {
+			isParentChainArbitrum = true
```

Comment:
This could fail due to other reasons like network issues. What I'd do is call client.GetCode on the arbsys address. If it's empty it's not an Arbitrum chain, if it's nonempty it's an Arbitrum Chain, and if it errors we can return that error.

Reply:
client does not have GetCode, I am assuming you meant AtCode, made changes according to that.